presto-main/src/main/java/com/facebook/presto/sql/planner/PlanOptimizersFactory.java

```diff
@@ -82,6 +83,7 @@ public PlanOptimizersFactory(Metadata metadata, SqlParser sqlParser, FeaturesCon
                 new CountConstantOptimizer(),
                 new WindowFilterPushDown(metadata), // This must run after PredicatePushDown and LimitPushDown so that it squashes any successive filter nodes and limits
                 new MergeProjections(),
+                new ImplementIntersectAsUnion(),
```

Comment:
This should probably be closer to the top of the list. I'd put it in the list after DesugaringOptimizer.